src/components/DateRangePicker.js

```diff
@@ -152,12 +156,49 @@ class DateRangePicker extends React.Component {
 
   _handleDefaultRangeSelection = (range) => {
     this.props.onDateSelect(range.getStartDate(), range.getEndDate(), range.displayValue);
+    this.setState({ focusedDay: range.getEndDate() });
 
     if (this.props.closeCalendarOnRangeSelect) {
       this._handleScrimClick();
     }
   };
 
+  _handleDayKeyDown = (e) => {
+    const startDate = moment.unix(this.state.currentDate).startOf('month').startOf('week');
+    const endDate = moment.unix(this.state.currentDate).endOf('month').endOf('week');
+
+    if (keycode(e) === 'right') {
+      if (moment.unix(this.state.focusedDay).add(1, 'days').startOf('day').isSameOrAfter(endDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).add(1, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).add(1, 'days').startOf('day').unix() });
+    } else if (keycode(e) === 'left') {
+      if (moment.unix(this.state.focusedDay).subtract(1, 'days').startOf('day').isBefore(startDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).subtract(1, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).subtract(1, 'days').startOf('day').unix() });
+    } else if (keycode(e) === 'enter') {
+      this._handleDateSelect(true, this.state.focusedDay);
+    } else if (keycode(e) === 'up') {
+      e.preventDefault(); //stop browserscrolling
+      if (moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day').isBefore(startDate)) {
+        this.setState({ currentDate: moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day').unix() });
+      }
+
+      this.setState({ focusedDay: moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day').unix() });
```

Comment:
Will you clean up the duplication in these `if` blocks so they're a little easier to read?
e.g.
```js
      const day = moment.unix(this.state.focusedDay).subtract(7, 'days').startOf('day');

      if (day.isBefore(startDate)) {
        this.setState({ currentDate: day.unix() });
      }

      this.setState({ focusedDay: day.unix() });
```

Reply:
yeah definitely. i also thought i made this a case/switch but i must have messed something up with my commits but let me put that back in as well

Reply:
ok i stuck with the if/else so i could block scope the `consts` which i think makes it cleaner and better than the switch/case anyway